src/influxDBTokenStreamHandler.py

```diff
@@ -80,13 +81,51 @@ def on_stream_closed(self) -> None:
         """
         logging.info('Subscribe to topic stream closed.')
 
-    def publish_response(self) -> None:
+    def get_publish_json(self, message):
+        """
+        Parse the correct token based on the IPC message received, and construct the final JSON to publish.
+
+        :param message: the received IPC messsage
+        :return: the complete JSON, including token, to publish
+        """
+
+        loaded_token_json = json.loads(self.influxDB_token_json)
+        publish_json = json.loads(self.influxDB_metadata_json)
+
+        if message['action'] == 'RetrieveToken':
```

Comment:
Re: our discussion offline:

it should always exist, since it’s constructed and sent by the influxdbpublisher. If it doesn’t exist, `getpublishjson()` will throw an exception that is caught by the caller, `handle_stream_event()`